### PR TITLE
[dx11] Add DX11 device interface definition

### PR DESCRIPTION
**Pull request by quadpixels**
Related issue = #992

This is the first part in an attempt to break down and refactor PR #3776.

During the refactor, a few things have changed (example: I would want to add a class to read information form the DX11 info queue for some functional testing), but it seems no matter what the refactor looks like, this interface boilerplate would be the same. So creating a pull request here.

### CODE REVIEW

**Review by bobcao3 — APPROVED** [review]
LGTM!

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import taichi.lang.linalg_impl
 import taichi.lang.meta
 from taichi._lib import core as _ti_core
+print(_ti_core.__file__)
```
Comment (ailzhang) [inline python/taichi/lang/__init__.py:19]: not necessary?

**Review by ailzhang — CHANGES_REQUESTED** [review]
Great work!
nit: please remove the additional print before merging :D

File: python/taichi/lang/__init__.py (hunk 1 above)
Comment (quadpixels) [reply python/taichi/lang/__init__.py:19]: This was added for resolving an installation issue, it is now removed.

**Comment by quadpixels** [comment]
> Great work! nit: please remove the additional print before merging :D

Thanks for the comments!
Removed the print statement in the new version of the pull request.

**Review by ailzhang — APPROVED** [review]
Thanks!

**Pull request merged**